pkg/registry/flowcontrol/bootstrap/default-objects.go

```diff
@@ -0,0 +1,270 @@
+	"k8s.io/apiserver/pkg/authentication/user"
+)
+
+// all the preserving objects
+var (
+	PreservingPriorityLevelConfigurations = []*rmv1a1.PriorityLevelConfiguration{
+		DefaultPriorityLevelConfigurationExempt,
+		DefaultPriorityLevelConfigurationDefault,
+	}
+	PreservingFlowSchemas = []*rmv1a1.FlowSchema{
+		DefaultFlowSchemaExempt,
+		DefaultFlowSchemaDefault,
+	}
+)
+
+// preserving default priority-levels
+var (
+	DefaultPriorityLevelConfigurationExempt = pl(
+		rmv1a1.PriorityLevelConfigurationNameExempt,
+		rmv1a1.PriorityLevelConfigurationSpec{Exempt: true},
+	)
+	DefaultPriorityLevelConfigurationDefault = pl(
+		"default",
+		rmv1a1.PriorityLevelConfigurationSpec{
+			AssuredConcurrencyShares: 100,
+			Queues:                   128,
+			HandSize:                 6,
+			QueueLengthLimit:         100,
+		})
+)
+
+// preserving default flow-schemas
+var (
+	DefaultFlowSchemaExempt = fs(
+		"exempt",
+		rmv1a1.PriorityLevelConfigurationNameExempt,
+		1500, "",
+		rmv1a1.PolicyRuleWithSubjects{
+			Subjects: groups(user.SystemPrivilegedGroup),
+			Rule: resourceRule(
+				[]string{rmv1a1.VerbAll},
+				[]string{rmv1a1.APIGroupAll},
+				[]string{rmv1a1.ResourceAll},
+			)},
+		rmv1a1.PolicyRuleWithSubjects{
+			Subjects: groups(user.SystemPrivilegedGroup),
+			Rule: nonResourceRule(
+				[]string{rmv1a1.VerbAll},
+				[]string{rmv1a1.NonResourceAll},
+			)},
+	)
+	DefaultFlowSchemaDefault = fs(
+		"default",
+		"default",
+		10000, rmv1a1.FlowDistinguisherMethodByUserType,
+		rmv1a1.PolicyRuleWithSubjects{
+			Subjects: groups(user.AllUnauthenticated, user.AllAuthenticated),
+			Rule: resourceRule(
+				[]string{rmv1a1.VerbAll},
+				[]string{rmv1a1.APIGroupAll},
+				[]string{rmv1a1.ResourceAll},
+			)},
+		rmv1a1.PolicyRuleWithSubjects{
+			Subjects: groups(user.AllUnauthenticated, user.AllAuthenticated),
+			Rule: nonResourceRule(
+				[]string{rmv1a1.VerbAll},
+				[]string{rmv1a1.NonResourceAll},
+			)},
+	)
+)
+
+// DefaultPriorityLevelConfigurations returns the instances of
+// the ones defined in the KEP, with system-top in position 0 and
+// workload-low in position 1
+func DefaultPriorityLevelConfigurations() []*rmv1a1.PriorityLevelConfiguration {
+	return []*rmv1a1.PriorityLevelConfiguration{
+		DefaultPriorityLevelConfigurationExempt,
+		pl("system", rmv1a1.PriorityLevelConfigurationSpec{
+			AssuredConcurrencyShares: 30,
+			Queues:                   1,
+			QueueLengthLimit:         1000}),
+		pl("workload-high", rmv1a1.PriorityLevelConfigurationSpec{
+			AssuredConcurrencyShares: 30,
+			Queues:                   128,
+			HandSize:                 6,
+			QueueLengthLimit:         100}),
+		pl("workload-low", rmv1a1.PriorityLevelConfigurationSpec{
+			AssuredConcurrencyShares: 30,
+			Queues:                   128,
+			HandSize:                 6,
+			QueueLengthLimit:         100}),
+		DefaultPriorityLevelConfigurationDefault,
+	}
+}
+
+// DefaultFlowSchemas returns instances of the FlowSchema
```

Comment:
I am not sure we want to auto-make these things, the first time is fine but we don't want fight with users after that, and I'm not sure how to avoid that. I wonder if it is better to publish recommended yamls and leave it up to distributions to actually solve the problem of installing and keeping these up to date?

@deads2k may have an idea?

Reply:
The KEP has a simple idea: at each apiserver startup, the predefined objects are created --- exempt PriorityLevelConfiguration last --- _if_ no exempt PriorityLevelConfiguration is found.

Reply:
There is a rewrite in https://github.com/kubernetes/enhancements/pull/1223 , perhaps that is easier to consume.